internal/resources/akscluster/resource_akscluster.go

```diff
@@ -0,0 +1,235 @@
+	if err := validate(nodepools); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := createOrUpdateCluster(data, tc.TMCConnection.AKSClusterResourceService); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := createNodepools(ctx, nodepools, tc.TMCConnection.AKSNodePoolResourceService); err != nil {
+		return diag.FromErr(err)
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, getTimeOut(data))
+	defer cancel()
+
+	if err := pollUntilReady(ctx, data, tc.TMCConnection, getPollInterval(ctx)); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return dataSourceTMCAKSClusterRead(ctx, data, tc)
+}
+
+// resourceClusterRead read state of existing AKS cluster and assigned nodepools.
+func resourceClusterRead(ctx context.Context, data *schema.ResourceData, config any) diag.Diagnostics {
+	return dataSourceTMCAKSClusterRead(ctx, data, config)
+}
+
+// resourceClusterInPlaceUpdate updates AKS cluster settings in place then manually reconciles nodepool changes
+// updating, creating, and deleting where appropriate.
+func resourceClusterInPlaceUpdate(ctx context.Context, data *schema.ResourceData, config any) diag.Diagnostics {
+	tc, ok := config.(authctx.TanzuContext)
+	if !ok {
+		return diag.Errorf("error while retrieving Tanzu auth config")
+	}
+
+	clusterResp, nodepoolResp, getErr := getClusterAndNodepools(ctx, data, tc.TMCConnection)
+	if getErr != nil || clusterResp == nil || nodepoolResp == nil {
+		return diag.FromErr(errors.Errorf("Unable to get Tanzu Mission Control AKS cluster entry, name : %s", data.Get(NameKey)))
+	}
+
+	// Make changes to the cluster config.
+	if clusterChange := data.HasChange("spec.0.config.0"); clusterChange {
+		if updateErr := updateClusterConfig(ctx, data, clusterResp, tc); updateErr != nil {
+			return diag.FromErr(updateErr)
+		}
+	}
+
+	// Make changes to cluster nodepools.
+	if data.HasChange("spec.0.nodepool") {
+		if npChangeErr := handleNodepoolChanges(ctx, nodepoolResp.Nodepools, data, tc.TMCConnection); npChangeErr != nil {
+			return diag.FromErr(npChangeErr)
+		}
+	}
+
+	return dataSourceTMCAKSClusterRead(ctx, data, config)
+}
+
+// resourceClusterDelete deletes an AKS cluster and all associated node pools.
+func resourceClusterDelete(ctx context.Context, data *schema.ResourceData, config any) diag.Diagnostics {
+	tc, ok := config.(authctx.TanzuContext)
+	if !ok {
+		return diag.Errorf("error while retrieving Tanzu auth config")
+	}
+
+	fn := extractClusterFullName(data)
+	if err := tc.TMCConnection.AKSClusterResourceService.AksClusterResourceServiceDelete(fn, "false"); err != nil && !clienterrors.IsNotFoundError(err) {
+		return diag.FromErr(errors.Wrapf(err, "Unable to delete Tanzu Mission Control AKS cluster entry, name : %s", data.Get(NameKey)))
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, getTimeOut(data))
+	defer cancel()
+
+	if err := pollUntilClusterDeleted(ctx, data, tc.TMCConnection.AKSClusterResourceService, getPollInterval(ctx)); err != nil {
+		return diag.FromErr(err)
+	}
```

Comment:
I think once the delete is complete the resource is removed from the state but better to add this explicitely
 `d.SetId("")` Ref: https://developer.hashicorp.com/terraform/tutorials/providers/provider-delete

Reply:
done